ios/FluentUI/Vnext/Avatar/AvatarGroup.swift

```diff
@@ -0,0 +1,172 @@
+        get {
+            return tokens.size
+        }
+        set {
+            tokens.size = newValue
+        }
+    }
+
+    var style: MSFAvatarGroupStyle {
+        get {
+            return tokens.style
+        }
+        set {
+            tokens.style = newValue
+        }
+    }
+
+    var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle,
+         size: MSFAvatarSize) {
+        self.tokens = MSFAvatarGroupTokens(style: style,
+                                      size: size)
+        super.init()
+    }
+}
+
+/// View that represents the AvatarGroup
+public struct AvatarGroup: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var state: MSFAvatarGroupStateImpl
+    @ObservedObject var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle, size: MSFAvatarSize) {
+        let state = MSFAvatarGroupStateImpl(style: style, size: size)
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        let size: CGFloat = tokens.size.size
+        let x: CGFloat = size + tokens.interspace - tokens.ringOuterGap - tokens.ringThickness
+        let avatars: [AvatarView] = state.avatars
+        let maxDisplayedAvatars: Int = avatars.prefix(Int(state.maxDisplayedAvatars)).count
+        let overflowCount: Int = Int((avatars.count > maxDisplayedAvatars ? UInt(avatars.count - maxDisplayedAvatars) : 0) + state.overflowCount)
+        HStack(spacing: 0) {
+            ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                let modify = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
+                state.avatars[index]
+                    .modifyIf(modify, { view in
+                        view.mask(AvatarCutout(xOrigin: x,
+                                               yOrigin: 0,
+                                               cutoutSize: size)
+                                    .fill(style: FillStyle(eoFill: true)))
+                    })
+                    .padding(.trailing, tokens.interspace)
+            }
+            if overflowCount > 0 {
+                createOverflow(count: overflowCount)
+            }
+        }
+    }
+
+    private func createOverflow(count: Int) -> AvatarView {
+        var avatar = AvatarView(style: .overflow, size: tokens.size)
+        let data = MSFAvatarStateImpl(style: .overflow, size: tokens.size)
+        data.primaryText = "\(count)"
+        data.image = nil
+        avatar.state = data
+        return avatar
+    }
+
+    /// Cutout shape for the succeeding Avatar in an Avatar Group in Stack style.
+    ///
+    /// xOrigin: beginning location of cutout on the x axis
+    ///
+    /// yOrigin: beginning location of cutout on the y axis
+    ///
+    /// cutoutSize: dimensions of cutout shape of the Avatar
+    private struct AvatarCutout: Shape {
```

Comment:
If I remember correctly, I asked you this on the previous PR:
In the scenario where the avatars have presence, does the cutout reflect that shape? (I think the same could be applicable in case the avatar is set for no transparency.

Reply:
The cutout is applied to the presence as well. Does this look ok?
<img width="202" alt="image" src="https://user-images.githubusercontent.com/22566866/121931467-a87f8680-cd11-11eb-9644-f6b80d33d034.png">

Reply:
doesn't make sense that avatar group has presence badge. the whole point of collaboration corner is to show the colorful ring based on activity.

Reply:
I think that supporting presence on the Avatar Pile style (where the Avatars don't overlap) will be nice. Not necessarily for the coauthoring scenario though.

But still it's worth testing the avatars with rings, setting with and without transparency for the Avatar Stack style.

Reply:
As I pointed in another comment, the cutout is incorrect when the rings are set to some avatars.

![Screen Shot 2021-06-25 at 10 05 38 AM](https://user-images.githubusercontent.com/68076145/123460709-e5167200-d59c-11eb-9270-0060f03f2850.png)